spec/nodeattr_interface_spec.rb

```diff
@@ -27,6 +27,23 @@
 require 'exceptions'
 
 RSpec.describe Metalware::NodeattrInterface do
+  context 'with setup1 genders' do
+    before do
+      SpecUtils.use_mock_genders(self, genders_file: 'setup1/genders')
+    end
+
+    describe '#nodes_in_primary_group' do
+      it 'returns only the nodes in the primary group' do
+        nodes = Metalware::NodeattrInterface.nodes_in_primary_group('group1')
+        nodes_not_in_primary_group = nodes.select do |node|
+          node.match(/\AnodeA\d\d\Z/).nil?
+        end
+        expect(nodes.length).to eq(10)
+        expect(nodes_not_in_primary_group.length).to eq(0)
```

Comment:
I still like the regex check as checking the length is shorter doesn't guarantee the correct nodes have been selected. However instead of using a negative match and confirming it has a length of 0, I have switched it to use the built in rspec `match` method. This should make it a bit clearer.